packages/design-system/src/components/Button/Button.tsx

```diff
@@ -42,156 +45,119 @@ type CommonButtonProps<T> = {
    */
   onClick?: (...args: any[]) => any;
   size?: ButtonSize;
-  /**
-   * Button [`type`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type) attribute
-   */
-  type?: ButtonType;
   /**
    * A string corresponding to the button-component variation classes.
    * The `'danger'` variation is deprecated and will be removed in a future release.
    */
   variation?: ButtonVariation;
 };
 
-type OmitProps = 'children' | 'className' | 'onClick' | 'ref' | 'size' | 'type' | 'href';
-
-type LinkButtonProps = CommonButtonProps<'a'> &
-  Omit<React.ComponentPropsWithRef<'a'>, OmitProps> & {
-    /**
-     * When provided the root component will render as an `<a>` element
-     * rather than `button`.
-     */
-    href?: string; // Still optional because it's optional on the anchor tag
-  };
-
-type ButtonButtonProps = CommonButtonProps<'button'> &
-  Omit<React.ComponentPropsWithRef<'button'>, OmitProps>;
-
-export type ButtonProps = CommonButtonProps<ButtonComponent> | LinkButtonProps | ButtonButtonProps;
-
-export class Button extends React.PureComponent<ButtonProps> {
-  static defaultProps = {
-    type: 'button',
```

Comment:
Not sure if there's an easy way to resolve this in the refactored component, but removing the default `type` for button could create a potential regression. In the case where a button is used inside of a form, the `type` should be specified to be `button` otherwise it will be defaulted to type=submit.

Not sure if this is a scenario in many of our consuming apps.

[From MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#notes)
```
If your buttons are not for submitting form data to a server, be sure to set their type attribute to button. Otherwise they will try to submit form data and to load the (nonexistent) response, possibly destroying the current state of the document.
```

Reply:
Ooh, good catch. I mistakenly thought they were optional 😬

Reply:
For some reason when I try adding the default props externally like 
```tsx
Button.defaultProps = {
  type: 'button',
}
```
I get some cryptic errors downstream in DrawerToggle. I think for functional components it's encouraged to define them as default parameters to the function, but I kind of wanted the default value to show up on the generated props table in our docs so we're transparent about `type` defaulting to `button`. Still fiddling with it...

Reply:
Pushed https://github.com/CMSgov/design-system/pull/1532/commits/79e3e9d3c192c50c1de7e76f7aa184f3ef316e61

Reply:
I think you can pass default props in the params you pass to the component, i.e., 
```js
export const Foo = ({bar, baz = "Default value"}) => {
    // logic
}
```

if `defaultProps` doesn't work in the future.